lib/cldr/export/data/numbers.rb

```diff
@@ -7,108 +7,181 @@ class Numbers < Base
-                format_result[format_key] = xpath_to_redirect(aliased.attribute("path").value)
+                pattern_result[pattern_key][pattern_count] = pattern_node.content
+              else
+                pattern_result[pattern_key] = pattern_node.content
               end
             end
           end
-
-          result[:default] = result[:default][:default] if result[:default]
           result
         end
 
-        def xpath_to_redirect(xpath)
-          length = xpath[/(\w+)FormatLength/, 1]
-          type   = xpath[/@type='(\w+)'/, 1]
-
-          :"numbers.formats.#{length}.patterns.#{type}"
+        def default_format_length_type
+          # TODO: It would be better is this were one of the valid values for the type attribute
+          # <!ATTLIST decimalFormatLength type (full | long | medium | short) #IMPLIED >
+          # But I haven't been able to figure out what the default is.
+          @default_format_length_type ||= :default
         end
 
-        def number_system(type)
-          node = select("numbers/#{type}Formats").first
-          begin
-            node.attribute("numberSystem").value
-          rescue
-            "latn"
+        def default_format_type
+          @default_format_type ||= begin
+            # Verify that the default format type has not changed / is the same for all the types
+            ldml_dtd_file = File.read("vendor/cldr/common/dtd/ldml.dtd")
+            ["currency", "decimal", "percent", "scientific"].each do |type|
+              next if ldml_dtd_file.include?("<!ATTLIST #{type}Format type NMTOKEN \"standard\" >")
+
+              raise "The default type for #{type}Format has changed. Some code will need to be updated."
+            end
+            :standard
           end
         end
 
-        def unit
-          @unit ||= select("numbers/currencyFormats/unitPattern").each_with_object({}) do |node, result|
-            count = node.attribute("count").value
+        def default_pattern_type
+          @default_pattern_type ||= begin
+            ldml_dtd_file = File.read("vendor/cldr/common/dtd/ldml.dtd")
+            ldml_dtd_file.match("<!ATTLIST pattern type NMTOKEN \"([^\"]+)\" >")[1]
+          end.to_sym
+        end
+
+        def xpath_to_default_currency_format_length_node_alias(xpath, number_system, format_length_key)
+          match = xpath.match(%r{\.\./currencyFormat\[@type='(\w+)+'\]})
+          raise "Alias doesn't match expected pattern: #{xpath}" unless match
+
+          target_type = match[1]
+          :"numbers.#{number_system}.formats.currency.patterns.#{format_length_key}.#{target_type}"
+        end
+
+        def xpath_to_redirect(xpath, number_system, length)
+          match = xpath.match(%r{\.\./#{length}FormatLength\[@type='(\w+)'\]})
+          raise "Alias doesn't match expected pattern: #{xpath}" unless match
+
+          type = match[1]
+          :"numbers.#{number_system}.formats.#{length}.patterns.#{type}"
+        end
```

Comment:
> because xxxFormatLength can exist without a type attribute
I think I see... even though there isn't an alias in `root` for `<xxxFormatLength>` without a `type`, in theory there could be, so perhaps we should support it??

At this point, each alias is implemented separately. Which aliases are supported is tracked by [`SUPPORTED_ALIAS_ELEMENT_CHAINS`](https://github.com/ruby-i18n/ruby-cldr/blob/6ac636c20920e3b7159ac890eb4f9b87220a0cd2/lib/cldr/validate_upstream_assumptions.rb#L32), so if any new ones were to be introduced, they would fail the `thor cldr:validate_upstream` checks done as part of upgrading the data.

(Which is to say, "We'll cross that bridge if/when we get to it.")